Remove nrfutil from requirements.txt

nordic-krch opened the pull request:
nrfutil is a python2 module and was placed in requirements.txt indended
to be used with python3. Created dedicated file for python2.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

thst-nordic commented:
There has been changes to the master branch Jenkinsfile.  

in order to get the 'CI/Jenkins/NRF' status you will need to rebase master.

nordic-krch commented:
> Shouldn't gs_ins_windows.rst get an update too, to add pip2 install scripts/requirements.txt or so?

done. Rebased to get the build green.

ru-fu commented:
Afaik, the installation (Zephyr installation instructions and the Getting Started Assistant) only install Python 3. So this must be updated as well if we require Python 2 to be installed in addition - and this will probably lead to path issues if the wrong one is in the path first.

nordic-krch commented:
@ru-fu it's only one script for shell over bluetooth that requires python 2. Maybe shouldn't then be mentioned in ncs getting started but only in https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/master/include/shell/shell_bt_nus.rst

ru-fu commented:
@nordic-krch That would definitely be easier than requiring it for everyone.

mbolivar commented:
@ru-fu @nordic-krch +1 to that idea assuming the updates to shell_bt_nus.rst support all platforms including Windows.

nordic-krch commented:
i've moved requirements file to `scripts/shell`. Added short note in `shell_bt_nus.rst`.

NordicBuilder commented:
**Some checks failed. Please fix and resubmit.**

## checkpatch issues

```
No additional types will be considered - file 'scripts/checkpatch/typedefsfile': No such file or directory
-:17: ERROR:TRAILING_WHITESPACE: trailing whitespace
#17: FILE: include/shell/shell_bt_nus.rst:44:
+   $

- total: 1 errors, 0 warnings, 22 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

NOTE: Whitespace errors detected.
      You may wish to use scripts/cleanpatch or scripts/cleanfile

Your patch has style problems, please review.

NOTE: Ignored message types: AVOID_EXTERNS BRACES CONFIG_EXPERIMENTAL CONST_STRUCT DATE_TIME FILE_PATH_CHANGES MINMAX MULTISTATEMENT_MACRO_USE_DO_WHILE NETWORKING_BLOCK_COMMENT_STYLE PRINTK_WITHOUT_KERN_LEVEL SPLIT_STRING VOLATILE

NOTE: If any of the errors are false positives, please report
      them to the maintainers.
```



*Tip: The bot edits this comment instead of posting a new one, so you can check the comment's history to see earlier messages.*

mbolivar commented on scripts/shell/requirements2.txt:
```diff
@@ -0,0 +1 @@
+nrfutil
```
nrfutil is going to be python 3 only very soon:

https://github.com/NordicSemiconductor/pc-nrfutil/issues/275#issuecomment-545335205

So I'm confused about why we're moving this particular dependency to our python 2 requirements file.

mbolivar replied:
Following up on this after some offline discussion with the nrfutil team regarding that issue:

- they are indeed going python 3, but it will actually be python 3.7 *only* (not 3.6, not 3.8, but exactly python 3.7)
- this means we should absolutely not have it in our top-level requirements.txt since we try to support as many python versions as Zephyr does (currently 3.4+)
- so I understand and agree that it makes sense to do this move, but since the python 3 nrfutil release is right around the corner, perhaps we can name this `scripts/shell/requirements.txt`, and document that for now this is python 2 but it will soon be python 3.7 only?

I don't think the nrfutil team is going to be able to support many versions of Python in the near future, so we may want to consider limiting our dependence on it in general-purpose NCS dependencies.

cc: @carlescufi

mbolivar replied:
Just saw this got a +1 on the documentation side. What are we going to do about the nrfutil question?

I noticed they have since put out another prerelease for 6.0, which will be python 3 only:

https://github.com/NordicSemiconductor/pc-nrfutil/releases

We could merge this as-is since the stable release is python 2 only, and update the documentation later once python 3 is available.

mbolivar replied:
nrfutil 6.0 is out: https://github.com/NordicSemiconductor/pc-nrfutil/issues/275#issuecomment-559777455

mbolivar replied:
However, note that it requires python 3.6+, while zephyr still claims support for 3.4+.

ru-fu commented on include/shell/shell_bt_nus.rst:
```diff
@@ -33,6 +33,15 @@ The script uses an additional Nordic Development Kit (for example, PCA10040) as
 It connects to the specified device and forwards all NUS traffic to the network port.
 You can then use a terminal application, for example PuTTY, to connect to that port (the default port is 8889) and use the shell.
 
+Script requires additional Python2 packages to be installed.
```
```suggestion
The script requires additional Python packages to be installed.
Currently, these packages are available for Python 2.x only.
```

nordic-krch commented:
I've updated commit, removed requirements file and added section in the `shell_bt_nus.rst` about nrfutil dependency. Eventually, `bt_nus_shell.py` should be ported to python 3. I've tried it but it requires more work since it includes porting to newer SD API.
